### ISSUE TITLE
New Rule: Report on unkown nodes

### ISSUE THREAD

**Issue by Corniel** [dotnet-project-file-analyzers/dotnet-project-file-analyzers#220]
Misspelled XML nodes/elements can lead to hard to track behavior. Tracking those unknown nodes is easy:

``` C#
protected override void Register(ProjectFileAnalysisContext<MsBuildProject> context)
{
    Walk(context.File, context);
}

private void Walk(Node node, ProjectFileAnalysisContext<MsBuildProject> context)
{
    if (node is Unknown)
    {
        context.ReportDiagnostic(Descriptor, node, node.LocalName);
    }

    foreach (var child in node.Children)
    {
        Walk(child, context);
    }
}
```

reporting on a node that is similar to a known node (hence a spelling/casing error) is also doable, and would lead to a rule with - at least I think - a lot of added value. 

Unfortunately, it is less straightforward to get a collection of supported nodes. A rule like this should allow to specify known elements, but still, without adding a lot of known nodes first, creating a rule like this would lead to a lot FP's. Any thoughts?

**Comment by Sjaaky** [comment]
The "list" of known types is here: https://learn.microsoft.com/en-us/visualstudio/msbuild/msbuild-project-file-schema-reference?view=vs-2022#msbuild-xsd-schema-definition-files
However, the possible values of OutputType are not defined. A few options are mentioned in the documentation.

**Comment by Corniel** [comment]
I also found this list: https://learn.microsoft.com/en-us/visualstudio/msbuild/common-msbuild-project-properties

But is this list long enough to report on all? Hoping that FP's are reported and added quickly? Or will users just be annoyed and disable the rule? (or even de-install the analyzers).

**Comment by CptWesley** [comment]
`IsTestProject` is already not among the lists, but is an uncommonly (on the scale of rare, uncommon, common) used node that is automatically inserted by the MsTest project template (and I believe it also has some interaction with Stryker.NET). We also use it for our test related analyzers

**Comment by Corniel** [comment]
@CptWesley Any node we have an analyzer for is known by our system, so those don't have to be added anymore.

**Comment by Corniel** [comment]
Due to the high change of reporting FP's, I'm gonna close this. I'd like the idea, but I think it will not fly.
